### PR TITLE
Implement Events blueprints and update Events tests

### PR DESCRIPTION
**Pull request by meshnesh**
# What does this PR do?

Implement separate Events blueprints and tests

## Description of tasks to be completed

Update tests for Events blueprints
Write tests for fetching events with no valid token
Create events blueprints and test they work
Update API documentation

## How should this be tested manually?
Run the application, test the api end points for all events
`/api/events/all`, requires no token
`/api/events/all/event_id`, requires no token
`/api/events`, requires a token, login to generate one
`/api/events/event_id`, requires a token, login to generate one, handles the PUT, DELETE methods.
`/api/events/event_id/rsvp`, requires a token, login to generate one, rsvp to an event
`/api/events/user_id`, requires a token, login to generate one, retrieves all events of a specific user

## Relevant tracker ID's
* 153231385
* 153729109
* 152980758
* 152975891
* 153729122

### CODE REVIEW

File: app/auth/views.py [hunk 1]
```diff
@@ -165,32 +183,36 @@ def get(self):
             resp = User.decode_token(access_token)
             if not isinstance(resp, str):
                 user = User.query.filter_by(id=resp).first()
-                responseObject = {
+                response_object = {
```
Comment (Rodgers-M) [inline app/auth/views.py:186]: good job folowing python naming conventions

File: app/events/views.py [hunk 2]
```diff
@@ -0,0 +1,378 @@
...
+                # Go ahead and handle the request, the user is authed
+
+                title = str(request.data.get('title', ''))
+                location = str(request.data.get('location', ''))
+                time = str(request.data.get('time', ''))
+                date = str(request.data.get('date', ''))
+                description = str(request.data.get('description', ''))
+                cartegory = str(request.data.get('cartegory', ''))
+                imageUrl = str(request.data.get('imageUrl', ''))
+                if title:
```
Comment (Rodgers-M) [inline app/events/views.py]: validate all fields, what if the `title` is present but `category` is missing?

**Review by Rodgers-M — COMMENTED** [review]
great job in handling the events' operations in their own blueprint.

**Comment by betsybeth** [comment]
Try and validate your arguements example when creating an event

**Comment by betsybeth** [comment]
Great work commenting your methods with docstrings

File: app/events/views.py [hunk 3]
```diff
@@ -0,0 +1,378 @@
+"""import depancies and methods."""
+
+from flask.views import MethodView
+from flask import make_response, request, jsonify
+from app.models import User, Events
+
+from . import events_blueprint
+
+
```
Comment (Kalela) [inline app/events/views.py:9]: Great work...I like your workflow. The abstractions make your code nice to read.

File: tests/test_event.py [hunk 4]
```diff
@@ -45,6 +45,39 @@ def login_user(self):
         }
         return self.client().post('/api/auth/login', data=user_data)
 
+    def test_api_get_all_events(self):
+        """Test API can get all events without token passed (GET request)."""
+        self.register_user()
```
Comment (frankip) [inline tests/test_event.py:51]: You can create a method for registering, logging in user and getting the access token

**Review by frankip — APPROVED** [review]
Great use of blueprint and following the PEP8 standards

File: app/events/views.py (hunk 3 above)
Comment (meshnesh) [reply app/events/views.py:9]: Thank you for the review

File: tests/test_event.py (hunk 4 above)
Comment (meshnesh) [reply tests/test_event.py:51]: I already have them at the top of the file.

Comment (frankip) [reply tests/test_event.py:51]: for the whole process of Registering, Loging In and getting access token and avoid repeating all the process every time you want to get a access token

File: app/events/views.py (hunk 3 above)
Comment (meshnesh) [reply app/events/views.py]: okay I now understand your review

File: tests/test_event.py (hunk 4 above)
Comment (meshnesh) [reply tests/test_event.py:51]: okay I now understand your review

**Pull request merged**